File: ProyectoEx/Usuario.cs
public class Usuario
{
    public Guid Id { get; private set; }
    public string Nombre { get; set; }
    public string Correo { get; set; }
    public string Telefono { get; set; }
    public string Calle { get; set; }
    public string FotoPerfil { get; set; }
    
    public List<Tarea> Tareas { get; set; }
    public Economia Economia { get; set; }
    public EstadoAnimo EstadoAnimo { get; set; }
    public Home Home { get; set; }

    public Usuario(string nombre, string correo, string telefono, string calle, string fotoPerfil)
    {
        Id = Guid.NewGuid();
        Nombre = nombre;
        Correo = correo;
        Telefono = telefono;
        Calle = calle;
        FotoPerfil = fotoPerfil;
        Tareas = new List<Tarea>();
        Economia = new Economia();
        EstadoAnimo = new EstadoAnimo();
        Home = new Home(this, nombre);
    }

    public void VerAjustes()
    {
        Console.WriteLine($"Mostrando ajustes de {Nombre}");
        Console.WriteLine($"Correo: {Correo}");
        Console.WriteLine($"Teléfono: {Telefono}");
        Console.WriteLine($"Dirección: {Calle}");
    }

    public void EditarPerfil(string nombre, string correo, string telefono, string calle, string fotoPerfil)
    {
        if (nombre != null) Nombre = nombre;
        if (correo != null) Correo = correo;
        if (telefono != null) Telefono = telefono;
        if (calle != null) Calle = calle;
        if (fotoPerfil != null) FotoPerfil = fotoPerfil;
        
        Console.WriteLine("Perfil actualizado correctamente");
    }
}

File: BocetoV2/Clases/Usuario.java
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.UUID;

public class Usuario {
    private String id;
    private String nombre;
    private String correo;
    private String telefono;
    private String calle;
    private String fotoPerfil;
    
    private List<Tarea> tareas;
    private Economia economia;
    private EstadoAnimo estadoAnimo;
    private Home home;

    public Usuario() {}

    public Usuario(String nombre, String correo, String telefono, String calle) {
        this.id = UUID.randomUUID().toString();
        this.nombre = nombre;
        this.correo = correo;
        this.telefono = telefono;
        this.calle = calle;
        this.tareas = new ArrayList<>();
        this.economia = new Economia();
        this.estadoAnimo = new EstadoAnimo();
        this.home = new Home(nombre);
    }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getNombre() { return nombre; }
    public void setNombre(String nombre) { this.nombre = nombre; }
    public String getCorreo() { return correo; }
    public void setCorreo(String correo) { this.correo = correo; }
    public String getTelefono() { return telefono; }
    public void setTelefono(String telefono) { this.telefono = telefono; }
    public String getCalle() { return calle; }
    public void setCalle(String calle) { this.calle = calle; }
    public String getFotoPerfil() { return fotoPerfil; }
    public void setFotoPerfil(String fotoPerfil) { this.fotoPerfil = fotoPerfil; }
    public List<Tarea> getTareas() { return tareas; }
    public void setTareas(List<Tarea> tareas) { this.tareas = tareas; }
    public Economia getEconomia() { return economia; }
    public void setEconomia(Economia economia) { this.economia = economia; }
    public EstadoAnimo getEstadoAnimo() { return estadoAnimo; }
    public void setEstadoAnimo(EstadoAnimo estadoAnimo) { this.estadoAnimo = estadoAnimo; }
    public Home getHome() { return home; }
    public void setHome(Home home) { this.home = home; }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Usuario usuario = (Usuario) o;
        return Objects.equals(id, usuario.id);
    }

    @Override
    public int hashCode() { return Objects.hash(id); }

    @Override
    public String toString() { return "Usuario{" + "nombre='" + nombre + '\'' + ", correo='" + correo + '\'' + '}'; }
}